### PR TITLE
WebView Dark theme support

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/414730916066338/1174433894299360/f
Tech Design URL: 
CC: 

**Description**:
This PR introduces new flags on the webview configuration to automatically use website dark them if DDG browser theme is Dark.

**Steps to test this PR**:
- Test:
1. Install the app
2. Visit SERP page
1. SERP should render in light mode
2. Change the app to use Dark theme
3. Go back to browser
4. Ensure SERP renders in dark mode
5. Go to SERP settings
6. Ensure dark theme is selected
7. Change the theme in SERP settings
8. Ensure selected theme is honored by webview
9. Now visit Twitter
10. Ensure it renders in dark mode
11. Change the app them to light
12. Ensure Twitter now is rendered in light theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Review by malmstein — APPROVED** [review]
works as expected, nice add @cmonfortep !

**Review by malmstein — CHANGES_REQUESTED** [review]
@cmonfortep after the changes introduced by https://github.com/duckduckgo/Android/pull/1398 this no longer works as expected. 

- Select system theme in Settings
- Select dark them in device settings
- WebView should be dark, but it's light (Twitter.com)

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 1]
```diff
@@ -1232,7 +1232,8 @@ class BrowserTabFragment :
 
     @SuppressLint("RequiresFeature")
     private fun configureDarkThemeSupport(webSettings: WebSettings) {
-        if (themingDataStore.theme == DuckDuckGoTheme.DARK) {
+        val currentNightMode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1235]: should be fixed now @malmstein

Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1235]: Actually, not yet. I will ping you once this is ready to review.

This is failing if you selected light theme but your device is in Dark mode.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 2]
```diff
@@ -1230,24 +1230,17 @@ class BrowserTabFragment :
         }
     }
 
-    @SuppressLint("RequiresFeature")
     private fun configureDarkThemeSupport(webSettings: WebSettings) {
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1233]: This is now ready for review @malmstein

**Review by malmstein — APPROVED** [review]
This is great, thanks @cmonfortep !

**Pull request merged**